### ISSUE TITLE
Can't parse cookies from set-cookie...

### ISSUE THREAD

**Issue by FlashThunder** [jshttp/cookie#50]
For example:

"NID=74=kU1rGWqE7kKmIrTtMVRJ-viL0fk1uih6tN0PTjiW2M_sh85c1WtSAsjuA3ITsx_mrcM37mya9N1HElQDeuM4siZb2Yn16cIE24i68YUsCLQpvsl-KPf3Rl7YmqwYDNRu; expires=Mon, 20-Jun-2016 19:36:37 GMT; path=/; domain=.google.pl; HttpOnly"

trying to parse above gives an error:

node_modules/cookie/index.js:49
    throw new TypeError('argument str must be a string');

**Comment by dougwilson** [comment]
Hi! That is correct: this module's `parse` function is only designed to parse the `Cookie` header, not the `Set-Cookie` header.

**Comment by wesleytodd** [comment]
Hey @dougwilson do you know of a module that does support parsing the `Set-Cookie` header?  

FWIW I also assumed it could parse the set header.  I will make a pull request to update the readme on this tonight, because I imagine it was a common assumption.

**Comment by dougwilson** [comment]
Rather than a pull request adding documentation about something this module has never done, I think a better approach would be to better describe what this module actually does. For example, I just looked at the readme and it wasn't really clear what header this parses regardless.

The reason I say this is because listing things something does not do is a slippery slope, one which can turn into an endless list. This module does a very poor job even describing what it is supposed to do, so I think that would be a better start.

In other words, I believe the assumption that this work on the set-cookie is just a symptom of the poor documentation, and just calling that out is only addressing the symptom, not the root cause.

**Comment by wesleytodd** [comment]
Yep, just like the docs I was unclear on my intentions. That is exactly
what I meant to document. I was also hoping to link to a module intended to
parse that header if you know one.

**Comment by wesleytodd** [comment]
For the discussion, is there a reason we don't add the ability to parse the `set-cookie` header to this module?  It seems like a logical place for it to live, especially since the serialize method is used to generate the `set-cookie` header value.

Maybe we could add an `extended: <Boolean>` option to the parse options which would also parse the extra params of the `set-cookie` header?
